coffeescripts/xwing.coffee

```diff
@@ -758,9 +763,22 @@ class exportObj.SquadBuilder
         $(window).resize =>
             @select_simple_view_button.click() if $(window).width() < 768 and @list_display_mode != 'simple'
 
-        @notes.change @onNotesUpdated
+        # separate function for on change that accesses the notes values
+        @notes.change @onNotesChanged
         @notes.on 'keyup', @onNotesUpdated
 
+    updatePermaLink: () =>
+        squad_link = "#{window.location.href.split('?')[0]}?f=#{encodeURI @faction}&d=#{encodeURI @serialize()}&sn=#{encodeURI @current_squad.name}"
```

Comment:
The squad name and notes should be encoded with `encodeURIComponent` (otherwise they don't properly handle URI characters like `#`).

...Come to think of it, I should have been using that instead of `encodeURI` all along...